sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -0,0 +1,256 @@
+package com.microsoft.azure.mobile.push;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import com.google.android.gms.gcm.GoogleCloudMessaging;
+import com.google.android.gms.iid.InstanceID;
+import com.microsoft.azure.mobile.AbstractMobileCenterService;
+import com.microsoft.azure.mobile.channel.Channel;
+import com.microsoft.azure.mobile.ingestion.models.json.LogFactory;
+import com.microsoft.azure.mobile.push.ingestion.models.PushInstallationLog;
+import com.microsoft.azure.mobile.push.ingestion.models.json.PushInstallationLogFactory;
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+import com.microsoft.azure.mobile.utils.storage.StorageHelper.PreferencesStorage;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Push notifications interface
+ */
+public class Push extends AbstractMobileCenterService {
+
+    /**
+     * Name of the service.
+     */
+    private static final String SERVICE_NAME = "Push";
+
+    /**
+     * TAG used in logging for Analytics.
+     */
+    public static final String LOG_TAG = MobileCenterLog.LOG_TAG + SERVICE_NAME;
+
+    /**
+     * Constant marking event of the push group.
+     */
+    private static final String PUSH_GROUP = "group_push";
+
+    /**
+     * Base key for stored preferences.
+     */
+    private static final String PREFERENCE_PREFIX = SERVICE_NAME + ".";
+
+    /**
+     * Preference key to store push token.
+     */
+    @VisibleForTesting
+    static final String PREFERENCE_KEY_PUSH_TOKEN = PREFERENCE_PREFIX + "push_token";
+
+
+    /**
+     * Shared instance.
+     */
+    @SuppressLint("StaticFieldLeak")
+    private static Push sInstance = null;
```

Comment:
In our coding conventions we don't initiliaze like that when its null, 0 or false.